superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -82,35 +76,31 @@ export default function AddDataset() {
     Reducer<Partial<DatasetObject> | null, DSReducerActionType>
   >(datasetReducer, null);
   const [hasColumns, setHasColumns] = useState(false);
-  const [datasets, setDatasets] = useState<DatasetObject[]>([]);
-  const datasetNames = datasets.map(dataset => dataset.table_name);
+  const [editPageIsVisible, setEditPageIsVisible] = useState(false);
   const encodedSchema = dataset?.schema
     ? encodeURIComponent(dataset?.schema)
     : undefined;
 
-  const getDatasetsList = useCallback(async () => {
+  const { getDatasetsList, datasets, datasetNames } = useGetDatasetsList();
```

Comment:
Non-blocking, but I think this hook could also be rewritten to encapsulate the get-dataset-list process a little better so there's less back-and-forth between the hook and the main function body.  I think you could rename it something like `useDatasetsList`, move the schema encoding and the effect inside of the hook, and have it return just `datasets` and `datasetNames`, so lines 80-96 would be replaced with a one-liner like this:
```javascript
  const { datasets, datasetNames } = useDatasetsList(dataset?.db, dataset?.schema);
```

Reply:
Same as above, this is much better! Updated in [`this commit`](https://github.com/apache/superset/pull/22043/commits/c6a9b0766df0321ff07e78973aeef82dc32ffbef).